Dockerfile

```diff
@@ -1,16 +1,32 @@
+## Build iPXE w/ IPv6 Support
+## Note: we are pinning to a specific commit for reproducible builds.
+## Updated as needed.
+FROM docker.io/centos:centos7 AS builder
+RUN yum install -y gcc git make genisoimage xz-devel
+WORKDIR /tmp
+COPY . .
+RUN git clone http://git.ipxe.org/ipxe.git && \
+      cd ipxe && \
+      git checkout 3fe683ebab29afacf224e6b0921f6329bebcdca7 && \
+      cd src && \
+      sed -i -e "s/#undef.*NET_PROTO_IPV6/#define NET_PROTO_IPV6/g" config/general.h && \
+      make bin/undionly.kpxe bin-x86_64-efi/ipxe.efi bin-x86_64-efi/snponly.efi
+
```

Comment:
These changes need to go into Dockerfile.ocp, unless we can get this into the downstream ipxe-bootimgs?

Reply:
yup, this may cause a problem, last time I tried to build a image in this build system with two "FROM " statements in the DockerFile  the build system changed one of them but not the other (this may be what you need or it might get in the way....), as far as I could see it happens here
https://github.com/openshift/builder/blob/master/pkg/build/builder/docker.go#L374

Reply:
I've asked:

Requesting IPv6 support: https://bugzilla.redhat.com/show_bug.cgi?id=1776926
Requesting snponly.efi: https://bugzilla.redhat.com/show_bug.cgi?id=1776929

Reply:
If it does the last FROM, it may not be a big issue as long as where it's built can hit docker.io to get the centos images, we only use the builder to make the ipxe images.

Reply:
1776926 has been marked as a dupe of https://bugzilla.redhat.com/show_bug.cgi?id=1637650 so perhaps we can use the non-qemu ipxe-roms package from the RHEL8 appstream repo?

Reply:
snpnoly.efi is still missing, which we need to make things work correctly.

Reply:
Ok I'll copy the current build-from-source for now and we can follow-up with a change that makes this work with the packaged ipxe-roms

Reply:
- the FROM that cahnges is just the last one (example for kuryr https://github.com/openshift/kuryr-kubernetes/blob/master/openshift-kuryr-cni-ci-rhel8.Dockerfile)
- the internal build infra can't hit docker.io or any other external site that is not explicitly allowed (we have the same issue for packages and source repos, which is managed directly in the openshift config)
as a workaround, we can have the source that we want to compile in the image repo itself or in a downstream repo
or, well, wait for the bugzilla to be resolved

Reply:
- the FROM that changes is just the last one (example for kuryr https://github.com/openshift/kuryr-kubernetes/blob/master/openshift-kuryr-cni-ci-rhel8.Dockerfile)
- the internal build infra can't hit docker.io or any other external site that is not explicitly allowed (we have the same issue for packages and source repos, which is managed directly in the openshift config)
as a workaround, we can have the source that we want to compile in the image repo itself or in a downstream repo
or, well, wait for the bugzilla to be resolved